be2-scala/src/main/scala/ch/epfl/pop/model/objects/RollCallData.scala

```diff
@@ -0,0 +1,32 @@
+package ch.epfl.pop.model.objects
+
+import ch.epfl.pop.json.MessageDataProtocol._
+import ch.epfl.pop.model.network.Parsable
+import ch.epfl.pop.model.network.method.message.Message
+import ch.epfl.pop.model.network.method.message.data.ActionType.ActionType
+import ch.epfl.pop.model.network.method.message.data.rollCall.{CloseRollCall, CreateRollCall, OpenRollCall, ReopenRollCall}
+import spray.json._
+
+final case class RollCallData(updateId: Hash, state: ActionType) {
+  def toJsonString: String = {
+    val that: RollCallData = this // tricks the compiler into inferring the right type
+    that.toJson.toString
+  }
+
+  def updateWith(message: Message): RollCallData = {
+    message.decodedData.fold(this) {
+      case create: CreateRollCall => RollCallData(create.id, create.action)
+      case open: OpenRollCall => RollCallData(open.update_id, open.action)
+      case reopen: ReopenRollCall => RollCallData(reopen.update_id, reopen.action)
+      case close: CloseRollCall => RollCallData(close.update_id, close.action)
+      case _ => this
+    }
```

Comment:
Took me a couple seconds to figure what that does. `Option.fold` works, but It seems like it would be clearer to just do

```suggestion
    message.decodedData match {
      case Some(create: CreateRollCall) => RollCallData(create.id, create.action)
      case Some(open: OpenRollCall) => RollCallData(open.update_id, open.action)
      case Some(reopen: ReopenRollCall) => RollCallData(reopen.update_id, reopen.action)
      case Some(close: CloseRollCall) => RollCallData(close.update_id, close.action)
      case _ => this
    }
```

Reply:
I wanted to be consistent with the implementation of the updateWith function of the `LaoData` (which uses the fold function), but if it seems cleaner, then I can change that.